### ISSUE TITLE
Refactor the test function module definitions

### ISSUE THREAD

**Issue by damar-wicaksono** [damar-wicaksono/uqtestfuns#100]
I think the way test functions are defined in each respective module can be improved.
Some of the obligatory module-level attributes are not that intuitive. For instance:

- `DEFAULT_INPUTS` would be better written as `AVAILABLE_INPUTS`
- `DEFAULT_PARAMETERS` would be better written as `AVAILABLE_PARAMETERS`

Some of the argument name in the top-level function `create_from_default()` can also be rethought.
For instance, `param_selection` and `input_selection` imply that the values are a keyword among available keywords.
Indeed this is how it is currently expected. However, I was thinking if the function can be called directly with a predefined input and parameters.

Furthermore, currently if `None` is passed to either of the arguments, then the default will be used. I was wondering whether `None` should indeed be `None` such these arguments can be set once an instance of `UQTestFuns` is created.
For instance:

```python
my_fun = uqtestfuns.create_from_default("sobol-g", input=my_custom_input, parameters=my_custom_parameters)
```

If custom input or parameters are given, the user should be responsible creating ones that are consistent with the evaluation function.

**Comment by damar-wicaksono** [comment]
Also, it may not be a good idea to instantiate the `MultivariateInput` in advance inside the test function modules because if the instance of the input is modified, one cannot revert back by calling the high-level function `create_from_default()`.

**Comment by damar-wicaksono** [comment]
This issue has been resolved by PR #113.

The following attributes now must be part of each of the test function module:

- `DEFAULT_NAME`
- `AVAILABLE_INPUT_SPECS`
- `DEFAULT_INPUT_SELECTION`
- `AVAILABLE_PARAMETERS`
- `DEFAULT_PARAMETERS_SELECTION`
- `evaluate()`
